### PR TITLE
Eliminate possible dangling/empty else condition

### PR DESCRIPTION
**Pull request by johnsonjh**
* Appease Oracle Lint by eliminating possible dangling/empty else condition

### CODE REVIEW

**Comment by aremmell** [comment]
So, what's the drill? Some compiler doesn't like that if-statement because finderr is only assigned to in certain blocks?

File: sirerrors.c [hunk 1]
```diff
@@ -106,15 +106,17 @@ void __sir_handleerr(int code, const char* func, const char* file, uint32_t line
...
              * error code.
              */
-#if defined(__HAVE_XSI_STRERROR_R__)
+# if defined(__HAVE_XSI_STRERROR_R__)
             _sir_selflog("strerror_r failed! error: %d", finderr);
-#elif defined(__HAVE_STRERROR_S__)
+# elif defined(__HAVE_STRERROR_S__)
             _sir_selflog("strerror_s failed! error: %d", finderr);
+# endif
 #endif
```
Comment (aremmell) [inline sirerrors.c:120]: Doesn't this #endif need to go outside that bracket?

**Comment by johnsonjh** [comment]
> So, what's the drill? Some compiler doesn't like that if-statement because finderr is only assigned to in certain blocks?

Oracle when being extra picky.

It doesn't like what turns out to be after the ifdef processing:

```
  if (blah) {
    Stuff;
  } else {}
```

File: sirerrors.c (hunk 1 above)
Comment (johnsonjh) [reply sirerrors.c:120]: image

So if neither defined, turns into:

image

Comment (aremmell) [reply sirerrors.c:120]: Oh, I missed the opener.

**Pull request merged**